src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked_skip_self_local_blocks/latest.sql

```diff
@@ -0,0 +1,57 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_isolation_test_session_is_blocked_skip_self_local_blocks(pBlockedPid integer, pInterestingPids integer[])
+RETURNS boolean AS $$
+  DECLARE
+    mBlockedGlobalPid int8;
+    workerProcessId integer := current_setting('citus.isolation_test_session_remote_process_id');
+    coordinatorProcessId integer := current_setting('citus.isolation_test_session_process_id');
+  BEGIN
+    IF pg_catalog.old_pg_isolation_test_session_is_blocked(pBlockedPid, pInterestingPids) THEN
+      RETURN true;
+    END IF;
+
+    -- pg says we're not blocked locally; check whether we're blocked globally.
+    -- Note that worker process may be blocked or waiting for a lock. So we need to
+    -- get transaction number for both of them. Following IF provides the transaction
+    -- number when the worker process waiting for other session.
+    IF EXISTS (SELECT 1 FROM get_global_active_transactions()
+               WHERE process_id = workerProcessId AND pBlockedPid = coordinatorProcessId) THEN
+      SELECT global_pid INTO mBlockedGlobalPid FROM get_global_active_transactions()
+      WHERE process_id = workerProcessId AND pBlockedPid = coordinatorProcessId;
+    ELSE
+      -- Check whether transactions initiated from the coordinator get locked
+      SELECT global_pid INTO mBlockedGlobalPid
+        FROM get_all_active_transactions() WHERE process_id = pBlockedPid;
+    END IF;
+
+    RETURN EXISTS (
+      SELECT 1 FROM citus_internal_global_blocked_processes()
+        WHERE waiting_global_pid = mBlockedGlobalPid
+        AND blocking_global_pid != waiting_global_pid
```

Comment:
as noted in the other [PR](https://github.com/citusdata/citus/pull/5910#discussion_r858416884), this might be a random local process as well, so we could consider adding:

> AND  blocking_global_pid != 0

Related to https://github.com/citusdata/citus/issues/1803. Deadlock detection and lock uses the same low level functions in Citus.

It becomes a lot less critical for deadlock detection because it requires *2* independent local processes involved. However, for `citus_lock_waits` or these UDFs, a single local process might trigger the false positive.